core/src/main/java/net/automatalib/automata/base/compact/AbstractCompactDeterministic.java

```diff
@@ -89,7 +89,11 @@ public Integer getInitialState() {
 
     @Override
     public T getTransition(int state, @Nullable I input) {
-        return getTransition(state, getSymbolIndex(input));
+        if (getInputAlphabet().containsSymbol(input)) {
+            return getTransition(state, getSymbolIndex(input));
+        } else {
+            return null;
+        }
```

Comment:
I reverted this change, and added a test case that now produces a NullPointerException in fcf164b.
In practice this NullPointerException occurs when the input alphabet from the automaton returned by ModelChecker#findCounterExample() is only a subset of the alphabet from the given automaton to that function.
So what happens is that e.g. an EmptinessOracle will try to obtain a word from the automaton that is returned by ModelChecker#findCounterExample() which will trigger the NullPointerException.

So there are several fixes:
1. Apply the fix in AbstractCompactDeterministic#getTransition().
1. Make ModelChecker#findCounterExample() also return an InputAlphabetHolder.
1. Make the FSM parser's parse() methods have the appropriate alphabet. This would require adding the original alphabet to the implementations of AbstractCompactDetermistic.

@mtf90 what do you think is the best solution?

Reply:
I think your testcase will always fail, because even if we fix the issue at hand (`findCounterExample` yields a "partial" automaton) the test will always provoke the "element not in the function's domain" error.

If I see things correctly, the `ModelChecker#findCounterExample()` method recieves a collection of input symbols which should be considered for counterexample search, right? I think it makes sense to expect that the returned counterexample can handle words over this alphabet. So it would be the easiest to simply pass this collection to the `parse` methods for constructing the `CompactDFA`s or `CompactMealy`s (so your third solution). It should be a small change to switch between "all the inputs in the .fsm file" vs. "I want the given input alphabet". I can write a few lines for this.

Reply:
I think the third solution is indeed best. If you know a good solution for adding it to the parse methods that would be great. Most of these DFA and Mealy parsers already have quite a few `parse()` methods to cope with all possible ways you could parse files. For example, you could use the same edge parser for both the input and output in case you parse a Mealy machine. I considered adding `Builders` for the constructors, but that does not allow you to cope with using the same edge parser I think.

Reply:
And the reason I committed the test cases is because they could be fixed by changing the `AbstractCompactDeterministic` class. Or by changing FSM parsers by adding an extra parameter with the input alphabet (the third solution) and then fixing the test case, by supplying this new alphabet.